cmd/krel/cmd/promote.go

```diff
@@ -0,0 +1,254 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"k8s.io/release/pkg/command"
+	"k8s.io/release/pkg/git"
+	"k8s.io/release/pkg/github"
+	"k8s.io/release/pkg/util"
+)
+
+const (
+	k8sioRepo             = "k8s.io"
+	k8sioManifestsPath    = "k8s.gcr.io"
+	stagingRepo           = "k8s-staging-kubernetes"
+	promotionBranchSuffix = "-image-promotion"
+)
+
+// promoteCommand is the krel subcommand to promote conainer images
+var promoteCommand = &cobra.Command{
+	Use:   "promote",
+	Short: "Starts an image promotion for a tag of kubernetes images",
+	Long: `krel promote
+
+The 'promote' subcommand of krel updates the image promoter manifests
+ans creates a PR in kubernetes/k8s.io`,
+	SilenceUsage:  true,
+	SilenceErrors: true,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		// Run the PR creation function
+		return runPromote()
+	},
+}
+
+type promoteOptions struct {
+	userFork        string
+	tag             string
+	interactiveMode bool
+}
+
+func (o *promoteOptions) Validate() error {
+	if o.tag == "" {
+		return errors.New("cannot start promotion --tag is required")
+	}
+	if o.userFork == "" {
+		return errors.New("cannot start promotion --fork is required")
+	}
+
+	// Check the fork slug
+	_, _, err := git.ParseRepoSlug(o.userFork)
+	if err != nil {
+		return errors.Wrap(err, "checking user's fork")
+	}
+
+	// Verify we got a valid tag
+	_, err = util.TagStringToSemver(o.tag)
+	if err != nil {
+		return errors.Wrapf(err, "verifying tag: %s", o.tag)
+	}
+
+	// Check that the GitHub token is set
+	token, isset := os.LookupEnv(github.TokenEnvKey)
+	if !isset || token == "" {
+		return errors.New("cannot promote images if GitHub token is not set")
+	}
+	return nil
+}
+
+var promoteOpts = &promoteOptions{}
+
+func init() {
+	promoteCommand.PersistentFlags().StringVarP(
+		&promoteOpts.tag,
+		"tag",
+		"t",
+		"",
+		"version tag of the images we will promote",
+	)
+
+	promoteCommand.PersistentFlags().StringVar(
+		&promoteOpts.userFork,
+		"fork",
+		"",
+		"the user's fork of kubernetes/k8s.io",
+	)
+
+	promoteCommand.PersistentFlags().BoolVarP(
+		&promoteOpts.interactiveMode,
+		"interactive",
+		"i",
+		false,
+		"interactive mode, asks before every step",
+	)
+
+	if err := promoteCommand.MarkPersistentFlagRequired("tag"); err != nil {
+		logrus.Error(errors.Wrap(err, "marking tag flag as required"))
+	}
+	if err := promoteCommand.MarkPersistentFlagRequired("fork"); err != nil {
+		logrus.Error(errors.Wrap(err, "marking tag flag as required"))
+	}
+
+	rootCmd.AddCommand(promoteCommand)
+}
+
+func runPromote() error {
+	// Validate options
+	branchname := promoteOpts.tag + promotionBranchSuffix
+
+	// Check the cmd line opts
+	if err := promoteOpts.Validate(); err != nil {
+		return errors.Wrap(err, "")
```

Comment:
This should include a message I guess :)

Reply:
Duh, yes I did not notice that one. Fixed.